lib/middleware.js

```diff
@@ -70,10 +71,14 @@ module.exports = function wrapper(context) {
         content = handleRangeHeaders(content, req, res);
 
         let contentType = mime.getType(filename);
-
-        // do not add charset to WebAssembly files, otherwise compileStreaming will fail in the client
-        if (!/\.wasm$/.test(filename)) {
-          contentType += '; charset=UTF-8';
+        const mimeInfo = mimeDb[contentType];
+        const charset = (mimeInfo && mimeInfo.charset)
+          // Special-cased text/html to prevent possible breaking change.
+          // TODO: remove the following line in next major release.
+          || (contentType === 'text/html' && 'UTF-8');
```

Comment:
`&& 'UTF-8'` always evaluates to `true`;

Reply:
It evaluates to the literal string 'UTF-8' to assign it to the charset variable.